File: resources/boilerplate/iOS-Native/java-JUnit5/iOSNative-oss.java
import io.appium.java_client.ios.IOSDriver;
import io.appium.java_client.ios.options.XCUITestOptions;
import org.junit.jupiter.api.*;
import org.openqa.selenium.By;

import java.net.MalformedURLException;
import java.net.URL;

public class LocaliOSTest {

    private IOSDriver driver;

    @BeforeEach
    public void setUp() throws MalformedURLException {
        XCUITestOptions options = new XCUITestOptions();
        options.setCapability("digitalai:testName", "[Enter Test Name here]");
        options.setCapability("digitalai:accessKey", "[Enter Your Access key here]");
        options.setCapability("digitalai:deviceQuery", "@os='ios' and @category='[Enter PHONE/TABLET here]'");
        options.setApp("cloud:com.experitest.ExperiBank");
        options.setBundleId("com.experitest.ExperiBank");
        driver = new IOSDriver(new URL("[Enter Instance here]/wd/hub"), options);
    }

    @Test
    public void quickStartiOSNativeDemo() {
        // [BEGIN_DEMO_STEPS]
        driver.findElement(By.xpath("//*[@name='usernameTextField']")).sendKeys("company");
        driver.findElement(By.xpath("//*[@name='passwordTextField']")).sendKeys("company");
        driver.findElement(By.xpath("//*[@name='loginButton']")).click();
        driver.findElement(By.xpath("//*[@name='makePaymentButton']")).click();
        driver.findElement(By.xpath("//*[@name='phoneTextField']")).sendKeys("0501234567");
        driver.findElement(By.xpath("//*[@name='nameTextField']")).sendKeys("John Snow");
        driver.findElement(By.xpath("//*[@name='amountTextField']")).sendKeys("50");
        driver.findElement(By.xpath("//*[@name='countryButton']")).click();
        driver.findElement(By.xpath("//*[@name='Switzerland']")).click();
        driver.findElement(By.xpath("//*[@name='sendPaymentButton']")).click();
        driver.findElement(By.xpath("//*[@name='Yes']")).click();
        // [END_DEMO_STEPS]
    }

    @AfterEach
    public void tearDown() {
        if (driver == null) return;
        try {
            System.out.println("Report URL: " + driver.getCapabilities().getCapability("digitalai:reportUrl"));
            System.out.println("Report Test ID: " + driver.getCapabilities().getCapability("digitalai:reportTestId"));
        } catch (Exception ignored) {}
        try {
            driver.quit();
        } catch (Exception ignored) {}
    }
}

File: resources/boilerplate/iOS-Native/Python/iOSNative-oss.py
import unittest
from appium import webdriver
from appium.options.common.base import AppiumOptions
from selenium.webdriver.common.by import By


class LocalIosTest(unittest.TestCase):
    testName = '[Enter TestName here]'
    accessKey = "[Enter Your Access key here]"
    driver = None

    def setUp(self):
        options = AppiumOptions()
        options.set_capability('digitalai:testName', self.testName)
        options.set_capability('digitalai:accessKey', self.accessKey)
        options.set_capability('app', 'cloud:com.experitest.ExperiBank')
        options.set_capability('bundleId', 'com.experitest.ExperiBank')
        options.set_capability('platformName', 'ios')
        options.set_capability('digitalai:deviceQuery', "@os='ios' and @category='[Enter PHONE/TABLET here]'")
        self.driver = webdriver.Remote("[Enter Instance here]/wd/hub", options=options)
        self.driver.implicitly_wait(10)

    def testQuickStartIosNativeDemo(self):
        # [BEGIN_DEMO_STEPS]
        self.driver.find_element(By.XPATH, "//*[@name='usernameTextField']").send_keys('company')
        self.driver.find_element(By.XPATH, "//*[@name='passwordTextField']").send_keys('company')
        self.driver.find_element(By.XPATH, "//*[@name='loginButton']").click()
        self.driver.find_element(By.XPATH, "//*[@name='makePaymentButton']").click()
        self.driver.find_element(By.XPATH, "//*[@name='phoneTextField']").send_keys('1234567')
        self.driver.find_element(By.XPATH, "//*[@name='nameTextField']").send_keys('Jon Snow')
        self.driver.find_element(By.XPATH, "//*[@name='amountTextField']").send_keys('50')
        self.driver.find_element(By.XPATH, "//*[@name='countryButton']").click()
        self.driver.find_element(By.XPATH, "//*[@name='Switzerland']").click()
        self.driver.find_element(By.XPATH, "//*[@name='sendPaymentButton']").click()
        self.driver.find_element(By.XPATH, "//*[@name='Yes']").click()
        # [END_DEMO_STEPS]

    def tearDown(self):
        print('Report URL: ' + self.driver.capabilities.get('digitalai:reportUrl', 'n/a'))
        self.driver.quit()


if __name__ == '__main__':
    unittest.main()
